compiler/src/iree/compiler/Dialect/LinalgExt/IR/TilingInterfaceImpl.cpp

```diff
@@ -2163,4 +2167,327 @@ LogicalResult OnlineAttentionOp::getResultTilePosition(
   return success();
 }
 
+//===---------------------------------------------------------------------===//
+// CustomOp
+//===---------------------------------------------------------------------===//
+
+/// These methods copied/modified from `TilingInterface` implementation of
+/// `getIterationDomain` of `LinalgOp`s.
+
+static SmallVector<OpFoldResult>
+createFlatListOfOperandDims(OpBuilder &builder, Location loc,
+                            CustomOp customOp) {
+  SmallVector<OpFoldResult> result;
+  for (Value operand : customOp->getOperands()) {
+    for (auto dim : llvm::seq<unsigned>(customOp.getRank(operand))) {
+      result.push_back(linalg::createFoldedDimOp(builder, loc, operand, dim));
+    }
+  }
+  return result;
+}
+
+/// Return dim expresssions that can be used as replacements in map that
+/// contains `numSymbols` symbols. The new dim expressions have positions
+/// `numDims, numDims + 1, numDims + 2, ...., numDims + numSymbols - 1`.
+static SmallVector<AffineExpr> getDimExprsForSymbols(MLIRContext *context,
+                                                     unsigned numDims,
+                                                     unsigned numSymbols) {
+  return llvm::map_to_vector(
+      llvm::seq<unsigned>(0, numSymbols), [&](unsigned symbolNumber) {
+        return getAffineDimExpr(symbolNumber + numDims, context);
+      });
+}
+
+/// Convert all symbols in the map to dim expressions, such that the new dim
+/// expressions have positions `numDims, numDims + 1, numDims + 2, ...., numDims
+/// + numSymbols - 1`.
+static AffineMap
+convertDimsToSymbols(AffineMap map, unsigned numDims, unsigned numSymbols,
+                     SmallVector<AffineExpr> &symbolReplacements) {
+  return map.replaceDimsAndSymbols(/*dimReplacements=*/ArrayRef<AffineExpr>{},
+                                   symbolReplacements, numDims + numSymbols, 0);
+}
+static SmallVector<AffineMap>
+convertDimsToSymbols(ArrayRef<AffineMap> maps, unsigned numDims,
+                     unsigned numSymbols,
+                     SmallVector<AffineExpr> &symbolReplacements) {
+  return llvm::map_to_vector(maps, [&](AffineMap map) {
+    return convertDimsToSymbols(map, numDims, numSymbols, symbolReplacements);
+  });
+}
+static SmallVector<AffineMap> convertDimsToSymbols(MLIRContext *context,
+                                                   ArrayRef<AffineMap> maps,
+                                                   unsigned numDims,
+                                                   unsigned numSymbols) {
+  auto symbolReplacements = getDimExprsForSymbols(context, numDims, numSymbols);
+  return convertDimsToSymbols(maps, numDims, numSymbols, symbolReplacements);
+}
+
+SmallVector<Range> CustomOp::getIterationDomainForDimensions(
```

Comment:
This implementation should be moved to `LinalgExtOps.cpp`.

Reply:
I dont know, the split of the `TilingInterfaceImpl` seems off to me. If it were an external implementation of the interface that would make sense, but this way the methods in two files is strange.

Reply:
I see. In this case, can you add a comment to `.td` which mentions that the implementation is in TilingInterfaceImpl? Moving the `getIterationDomain` to `IR/LinalgExtOps.cpp` looks weird to me..